src/lib/quoteEngine.js

```diff
@@ -64,47 +70,17 @@ const calculateOptimalPoolAllocation = (coverAmount, pools) => {
   return allocations;
 };
 
-/**
- * Allocates a given amount to pools based on a custom pool ID priority list.
- *
- * @param {number} amountToAllocate - The amount to be allocated.
- * @param {Array<object>} poolsData - An array of pool data objects.
- * @param {Array<number>} customPoolIdPriority - A list of pool IDs in priority order.
- * @return {object} An object containing the allocations by pool ID. (poolId => BigNumber amount)
- */
-const customAllocationPriorityFixedPrice = (amountToAllocate, poolsData, customPoolIdPriority) => {
-  const allocations = {};
-  let coverAmountLeft = amountToAllocate;
-  const customPoolIdPriorityCopy = customPoolIdPriority.slice(0); // avoid mutation on the customPoolIdPriority array
-
-  while (coverAmountLeft > 0 && customPoolIdPriorityCopy.length > 0) {
-    const poolId = customPoolIdPriorityCopy.shift();
-    const pool = poolsData.find(poolData => poolData.poolId === poolId);
-    if (!pool) {
-      console.warn(`Unable to find pool ${poolId} in poolsData array. Skipping\n`);
-      console.warn(`Available poolIds in poolsData: ${poolsData.map(p => p.poolId).join(', ')}`);
-      console.debug('poolsData: ', inspect(poolsData, { depth: null }));
-      continue;
-    }
-
-    let availableCapacity = pool.totalCapacity.sub(pool.initialCapacityUsed);
+function sortPools(poolsData, customPoolIdPriorityFixedPrice) {
+  const poolIdsByPrice = poolsData.sort((a, b) => a.basePrice - b.basePrice).map(p => p.poolId);
+  const allPoolIds = poolsData.map(p => p.poolId);
```

Comment:
(nit) it would be more performant if this is a map/object instead of an array

Reply:
we have like 20 pools, there's no perf issue to fix